app/src/ui/history/compare.tsx

```diff
@@ -0,0 +1,478 @@
+  readonly repository: Repository
+  readonly compareState: ICompareState
+  readonly gitHubUsers: Map<string, IGitHubUser>
+  readonly emoji: Map<string, string>
+  readonly commitLookup: Map<string, Commit>
+  readonly localCommitSHAs: ReadonlyArray<string>
+  readonly dispatcher: Dispatcher
+  readonly currentBranch: Branch | null
+  readonly sidebarHasFocusWithin: boolean
+  readonly onRevertCommit: (commit: Commit) => void
+  readonly onViewCommitOnGitHub: (sha: string) => void
+}
+
+interface ICompareSidebarState {
+  /**
+   * This branch should only be used when tracking interactions that the user is performing.
+   *
+   * For all other cases, use the prop
+   */
+  readonly focusedBranch: Branch | null
+  readonly filterText: string
+  readonly showBranchList: boolean
+  readonly selectedCommit: Commit | null
+}
+
+/** If we're within this many rows from the bottom, load the next history batch. */
+const CloseToBottomThreshold = 10
+
+export class CompareSidebar extends React.Component<
+  ICompareSidebarProps,
+  ICompareSidebarState
+> {
+  private textbox: TextBox | null = null
+  private readonly loadChangedFilesScheduler = new ThrottledScheduler(200)
+  private branchList: BranchList | null = null
+
+  public constructor(props: ICompareSidebarProps) {
+    super(props)
+
+    this.state = {
+      focusedBranch: null,
+      filterText: '',
+      showBranchList: false,
+      selectedCommit: null,
+    }
+  }
+
+  public componentWillReceiveProps(nextProps: ICompareSidebarProps) {
+    const hasFormStateChanged =
+      nextProps.compareState.formState.kind !==
+      this.props.compareState.formState.kind
+
+    const newFormState = nextProps.compareState.formState
+
+    if (hasFormStateChanged && newFormState.kind === ComparisonView.None) {
+      // the comparison form should be reset to its default state
+      this.setState({ filterText: '', focusedBranch: null })
+      return
+    }
+
+    if (!hasFormStateChanged && newFormState.kind !== ComparisonView.None) {
+      // ensure the filter text is in sync with the comparison branch
+      const branch = newFormState.comparisonBranch
+
+      this.setState({
+        filterText: branch.name,
+        focusedBranch: branch,
+      })
+    }
+
+    if (nextProps.sidebarHasFocusWithin !== this.props.sidebarHasFocusWithin) {
+      if (nextProps.sidebarHasFocusWithin === false) {
+        this.setState({ showBranchList: false })
+      }
+    }
+  }
+
+  public componentWillMount() {
+    this.props.dispatcher.initializeCompare(this.props.repository)
+  }
+
+  public componentWillUnmount() {
+    this.textbox = null
+  }
+
+  public componentDidMount() {
+    if (this.textbox !== null && this.state.showBranchList) {
+      this.textbox.focus()
+    }
+  }
+
+  public render() {
+    const formState = this.props.compareState.formState
+    const placeholderText =
+      formState.kind === ComparisonView.None
+        ? __DARWIN__
+          ? 'Select Branch To Compare...'
```

Comment:
We're not 💯 consistent but usually small conjunction words like "in" and "to" and "and" are lowercase (cross-checked with the menu bar). If we follow suit, this should be:
```js
? 'Select Branch to Compare...'
```